stdlib/Profile/test/runtests.jl

```diff
@@ -276,14 +276,14 @@ end
 @testset "HeapSnapshot" begin
     tmpdir = mktempdir()
     fname = cd(tmpdir) do
-        read(`$(Base.julia_cmd()) --startup-file=no -e "using Profile; print(Profile.take_heap_snapshot())"`, String)
+        read(`$(Base.julia_cmd()) --startup-file=no -e "using Profile; const x = \"redact_this\"; print(Profile.take_heap_snapshot())"`, String)
     end
 
     @test isfile(fname)
 
-    open(fname) do fs
-        @test readline(fs) != ""
-    end
+    sshot = read(fname, String)
+    @test sshot != ""
+    @test !contains(sshot, "redact_this")
```

Comment:
Can you please add another test that if you set `redact_data=false`, it _does_ have the string data? 🙏

Reply:
👍